{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "panel_list = [fname for fname in os.listdir('../data/crc_raw/cBioPortal_files/') if 'data_gene_panel_' in fname]\n",
    "len(panel_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_path = '../data/crc_raw/cBioPortal_files/'\n",
    "panel_dict = {}\n",
    "for panel in panel_list:\n",
    "    panel_dict[panel] = pd.read_csv(data_path + panel, sep='\\t', skiprows=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "col_dict = {}\n",
    "for panel in panel_list:\n",
    "    col_dict[panel] = panel_dict[panel].columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "799"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#make a list of all the genes in all the panels\n",
    "gene_list = []\n",
    "for panel in panel_list:\n",
    "    gene_list.extend(list(panel_dict[panel].columns))\n",
    "gene_list = list(set(gene_list))\n",
    "len(gene_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAigAAAGdCAYAAAA44ojeAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/bCgiHAAAACXBIWXMAAA9hAAAPYQGoP6dpAAAev0lEQVR4nO3de3BUhdnH8V8uJlzMJg2SbFJCuHgJdylgXFHEkgIhpTJSKzYqAgNTZ6NCKkKsguIl6DhqtQjiWNApEbUVLGmNQpCkDAFClApoI1AqKGxwpGQhDgGSff9o3b4reAkme54k38/MmeFcsvucDOrXs2d3IwKBQEAAAACGRDo9AAAAwFcRKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADAn2ukBzkVjY6MOHjyouLg4RUREOD0OAAD4DgKBgI4dO6bU1FRFRn7zNZJWGSgHDx5UWlqa02MAAIBzcODAAXXr1u0bj2mVgRIXFyfpPyfocrkcngYAAHwXfr9faWlpwf+Of5NWGShfvqzjcrkIFAAAWpnvcnsGN8kCAABzCBQAAGAOgQIAAMwhUAAAgDkECgAAMIdAAQAA5hAoAADAHAIFAACYQ6AAAABzCBQAAGAOgQIAAMwhUAAAgDkECgAAMIdAAQAA5kQ7PYBFPeb+JWT9XwtzHJoEAID2iSsoAADAHAIFAACY06RAKSws1LBhwxQXF6ekpCRNmDBB1dXVIceMHDlSERERIcuvfvWrkGP279+vnJwcderUSUlJSZo9e7ZOnz79/c8GAAC0CU26B6WsrExer1fDhg3T6dOndc8992j06NH64IMP1Llz5+Bx06dP14IFC4LrnTp1Cv65oaFBOTk5crvd2rRpkw4dOqRbbrlF5513nh555JFmOCUAANDaNSlQSkpKQtaXL1+upKQkVVVVacSIEcHtnTp1ktvtPutjvP322/rggw+0bt06JScn69JLL9WDDz6oOXPm6P7771dMTMw5nAYAAGhLvtc9KLW1tZKkxMTEkO0rVqzQBRdcoP79+6ugoEBffPFFcF9FRYUGDBig5OTk4LYxY8bI7/dr165dZ32e+vp6+f3+kAUAALRd5/w248bGRs2cOVPDhw9X//79g9t/+ctfKj09XampqXr//fc1Z84cVVdX6/XXX5ck+Xy+kDiRFFz3+Xxnfa7CwkI98MAD5zoqAABoZc45ULxer3bu3KmNGzeGbJ8xY0bwzwMGDFBKSopGjRqlvXv3qnfv3uf0XAUFBcrPzw+u+/1+paWlndvgAADAvHN6iScvL0/FxcV655131K1bt288NjMzU5K0Z88eSZLb7VZNTU3IMV+uf919K7GxsXK5XCELAABou5oUKIFAQHl5eVq1apXWr1+vnj17fuvPbN++XZKUkpIiSfJ4PNqxY4cOHz4cPGbt2rVyuVzq27dvU8YBAABtVJNe4vF6vSoqKtIbb7yhuLi44D0j8fHx6tixo/bu3auioiKNGzdOXbp00fvvv69Zs2ZpxIgRGjhwoCRp9OjR6tu3r26++WY99thj8vl8uvfee+X1ehUbG9v8ZwgAAFqdJl1BWbx4sWprazVy5EilpKQEl1deeUWSFBMTo3Xr1mn06NHKyMjQr3/9a02cOFFr1qwJPkZUVJSKi4sVFRUlj8ejm266SbfcckvI56YAAID2rUlXUAKBwDfuT0tLU1lZ2bc+Tnp6uv7617825akBAEA7wnfxAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgTpMCpbCwUMOGDVNcXJySkpI0YcIEVVdXhxxz4sQJeb1edenSReeff74mTpyompqakGP279+vnJwcderUSUlJSZo9e7ZOnz79/c8GAAC0CU0KlLKyMnm9Xm3evFlr167VqVOnNHr0aNXV1QWPmTVrltasWaPXXntNZWVlOnjwoK677rrg/oaGBuXk5OjkyZPatGmTXnzxRS1fvlzz5s1rvrMCAACtWkQgEAic6w9/9tlnSkpKUllZmUaMGKHa2lp17dpVRUVF+vnPfy5J+sc//qE+ffqooqJCl19+ud5880399Kc/1cGDB5WcnCxJWrJkiebMmaPPPvtMMTEx3/q8fr9f8fHxqq2tlcvlOtfxv1aPuX8JWf/Xwpxmfw4AANqbpvz3+3vdg1JbWytJSkxMlCRVVVXp1KlTysrKCh6TkZGh7t27q6KiQpJUUVGhAQMGBONEksaMGSO/369du3ad9Xnq6+vl9/tDFgAA0Hadc6A0NjZq5syZGj58uPr37y9J8vl8iomJUUJCQsixycnJ8vl8wWP+f5x8uf/LfWdTWFio+Pj44JKWlnauYwMAgFbgnAPF6/Vq586dWrlyZXPOc1YFBQWqra0NLgcOHGjx5wQAAM6JPpcfysvLU3FxscrLy9WtW7fgdrfbrZMnT+ro0aMhV1FqamrkdruDx2zdujXk8b58l8+Xx3xVbGysYmNjz2VUAADQCjXpCkogEFBeXp5WrVql9evXq2fPniH7hwwZovPOO0+lpaXBbdXV1dq/f788Ho8kyePxaMeOHTp8+HDwmLVr18rlcqlv377f51wAAEAb0aQrKF6vV0VFRXrjjTcUFxcXvGckPj5eHTt2VHx8vKZNm6b8/HwlJibK5XLp9ttvl8fj0eWXXy5JGj16tPr27aubb75Zjz32mHw+n+699155vV6ukgAAAElNDJTFixdLkkaOHBmyfdmyZbr11lslSU8++aQiIyM1ceJE1dfXa8yYMXr22WeDx0ZFRam4uFi33XabPB6POnfurMmTJ2vBggXf70wAAECb8b0+B8UpfA4KAACtT9g+BwUAAKAlECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMCcJgdKeXm5xo8fr9TUVEVERGj16tUh+2+99VZFRESELGPHjg055siRI8rNzZXL5VJCQoKmTZum48ePf68TAQAAbUeTA6Wurk6DBg3SokWLvvaYsWPH6tChQ8Hl5ZdfDtmfm5urXbt2ae3atSouLlZ5eblmzJjR9OkBAECbFN3UH8jOzlZ2dvY3HhMbGyu3233WfR9++KFKSkpUWVmpoUOHSpKeeeYZjRs3To8//rhSU1ObOhIAAGhjWuQelA0bNigpKUmXXHKJbrvtNn3++efBfRUVFUpISAjGiSRlZWUpMjJSW7ZsOevj1dfXy+/3hywAAKDtavZAGTt2rF566SWVlpbq0UcfVVlZmbKzs9XQ0CBJ8vl8SkpKCvmZ6OhoJSYmyufznfUxCwsLFR8fH1zS0tKae2wAAGBIk1/i+TaTJk0K/nnAgAEaOHCgevfurQ0bNmjUqFHn9JgFBQXKz88Prvv9fiIFAIA2rMXfZtyrVy9dcMEF2rNnjyTJ7Xbr8OHDIcecPn1aR44c+dr7VmJjY+VyuUIWAADQdrV4oHzyySf6/PPPlZKSIknyeDw6evSoqqqqgsesX79ejY2NyszMbOlxAABAK9Dkl3iOHz8evBoiSfv27dP27duVmJioxMREPfDAA5o4caLcbrf27t2ru+++WxdeeKHGjBkjSerTp4/Gjh2r6dOna8mSJTp16pTy8vI0adIk3sEDAAAkncMVlG3btmnw4MEaPHiwJCk/P1+DBw/WvHnzFBUVpffff18/+9nPdPHFF2vatGkaMmSI/va3vyk2Njb4GCtWrFBGRoZGjRqlcePG6corr9TSpUub76wAAECr1uQrKCNHjlQgEPja/W+99da3PkZiYqKKioqa+tQAAKCd4Lt4AACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwp8kfdY+2rcfcv4Ss/2thjkOTAADaM66gAAAAcwgUAABgDoECAADMIVAAAIA53CQLAIB4k4A1XEEBAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMAcAgUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAOQQKAAAwh0ABAADmECgAAMCcJgdKeXm5xo8fr9TUVEVERGj16tUh+wOBgObNm6eUlBR17NhRWVlZ2r17d8gxR44cUW5urlwulxISEjRt2jQdP378e50IAABoO5ocKHV1dRo0aJAWLVp01v2PPfaYnn76aS1ZskRbtmxR586dNWbMGJ04cSJ4TG5urnbt2qW1a9equLhY5eXlmjFjxrmfBQAAaFOim/oD2dnZys7OPuu+QCCgp556Svfee6+uvfZaSdJLL72k5ORkrV69WpMmTdKHH36okpISVVZWaujQoZKkZ555RuPGjdPjjz+u1NTU73E6AACgLWjWe1D27dsnn8+nrKys4Lb4+HhlZmaqoqJCklRRUaGEhIRgnEhSVlaWIiMjtWXLlrM+bn19vfx+f8gCAADarmYNFJ/PJ0lKTk4O2Z6cnBzc5/P5lJSUFLI/OjpaiYmJwWO+qrCwUPHx8cElLS2tOccGAADGtIp38RQUFKi2tja4HDhwwOmRAABAC2rWQHG73ZKkmpqakO01NTXBfW63W4cPHw7Zf/r0aR05ciR4zFfFxsbK5XKFLAAAoO1q1kDp2bOn3G63SktLg9v8fr+2bNkij8cjSfJ4PDp69KiqqqqCx6xfv16NjY3KzMxsznEAAEAr1eR38Rw/flx79uwJru/bt0/bt29XYmKiunfvrpkzZ+qhhx7SRRddpJ49e+q+++5TamqqJkyYIEnq06ePxo4dq+nTp2vJkiU6deqU8vLyNGnSJN7BAwAAJJ1DoGzbtk3XXHNNcD0/P1+SNHnyZC1fvlx333236urqNGPGDB09elRXXnmlSkpK1KFDh+DPrFixQnl5eRo1apQiIyM1ceJEPf30081wOgAAoC1ocqCMHDlSgUDga/dHRERowYIFWrBgwdcek5iYqKKioqY+NQAAaCdaxbt4AABA+0KgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgTrTTAwBW9Jj7l5D1fy3McWgSAABXUAAAgDkECgAAMIdAAQAA5hAoAADAHAIFAACYQ6AAAABzCBQAAGAOgQIAAMwhUAAAgDkECgAAMIdAAQAA5vBdPACAIL6TClZwBQUAAJhDoAAAAHMIFAAAYA6BAgAAzCFQAACAObyLB8AZ79yQePcGAGdxBQUAAJhDoAAAAHMIFAAAYA73oADAV/BpqoDzmv0Kyv3336+IiIiQJSMjI7j/xIkT8nq96tKli84//3xNnDhRNTU1zT0GAABoxVrkJZ5+/frp0KFDwWXjxo3BfbNmzdKaNWv02muvqaysTAcPHtR1113XEmMAAIBWqkVe4omOjpbb7T5je21trV544QUVFRXpxz/+sSRp2bJl6tOnjzZv3qzLL7+8JcYBAACtTItcQdm9e7dSU1PVq1cv5ebmav/+/ZKkqqoqnTp1SllZWcFjMzIy1L17d1VUVHzt49XX18vv94csAACg7Wr2QMnMzNTy5ctVUlKixYsXa9++fbrqqqt07Ngx+Xw+xcTEKCEhIeRnkpOT5fP5vvYxCwsLFR8fH1zS0tKae2wAAGBIs7/Ek52dHfzzwIEDlZmZqfT0dL366qvq2LHjOT1mQUGB8vPzg+t+v59IAQCgDWvxz0FJSEjQxRdfrD179sjtduvkyZM6evRoyDE1NTVnvWflS7GxsXK5XCELAABou1o8UI4fP669e/cqJSVFQ4YM0XnnnafS0tLg/urqau3fv18ej6elRwEAAK1Es7/Ec9ddd2n8+PFKT0/XwYMHNX/+fEVFRenGG29UfHy8pk2bpvz8fCUmJsrlcun222+Xx+PhHTwAACCo2QPlk08+0Y033qjPP/9cXbt21ZVXXqnNmzera9eukqQnn3xSkZGRmjhxourr6zVmzBg9++yzzT0GAABoxZo9UFauXPmN+zt06KBFixZp0aJFzf3UAACgjeDLAgEAgDkECgAAMIdAAQAA5hAoAADAHAIFAACYQ6AAAABzCBQAAGAOgQIAAMwhUAAAgDkECgAAMIdAAQAA5hAoAADAHAIFAACYQ6AAAABzCBQAAGAOgQIAAMwhUAAAgDkECgAAMIdAAQAA5hAoAADAHAIFAACYE+30AAAAwFk95v7ljG3/WpjjwCT/wxUUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADCHQAEAAOYQKAAAwBwCBQAAmEOgAAAAcwgUAABgDoECAADMIVAAAIA5BAoAADDH0UBZtGiRevTooQ4dOigzM1Nbt251chwAAGCEY4HyyiuvKD8/X/Pnz9e7776rQYMGacyYMTp8+LBTIwEAACMcC5QnnnhC06dP15QpU9S3b18tWbJEnTp10u9//3unRgIAAEZEO/GkJ0+eVFVVlQoKCoLbIiMjlZWVpYqKijOOr6+vV319fXC9trZWkuT3+1tkvsb6L0LWW+p5LOLc/6c9n7vUvs7/q/i78D+ce/sQrn8HfPmYgUDg2w8OOODTTz8NSAps2rQpZPvs2bMDl1122RnHz58/PyCJhYWFhYWFpQ0sBw4c+NZWcOQKSlMVFBQoPz8/uN7Y2KgjR46oS5cuioiIaNbn8vv9SktL04EDB+RyuZr1sVsDzr99n7/E76C9n7/E76C9n7/Ucr+DQCCgY8eOKTU19VuPdSRQLrjgAkVFRammpiZke01Njdxu9xnHx8bGKjY2NmRbQkJCS44ol8vVbv9iSpx/ez9/id9Bez9/id9Bez9/qWV+B/Hx8d/pOEduko2JidGQIUNUWloa3NbY2KjS0lJ5PB4nRgIAAIY49hJPfn6+Jk+erKFDh+qyyy7TU089pbq6Ok2ZMsWpkQAAgBGOBcoNN9ygzz77TPPmzZPP59Oll16qkpISJScnOzWSpP+8nDR//vwzXlJqLzj/9n3+Er+D9n7+Er+D9n7+ko3fQUQg8F3e6wMAABA+fBcPAAAwh0ABAADmECgAAMAcAgUAAJhDoPxXeXm5xo8fr9TUVEVERGj16tVOjxRWhYWFGjZsmOLi4pSUlKQJEyaourra6bHCZvHixRo4cGDwQ4k8Ho/efPNNp8dyzMKFCxUREaGZM2c6PUrY3H///YqIiAhZMjIynB4rrD799FPddNNN6tKlizp27KgBAwZo27ZtTo8VNj169Djj70BERIS8Xq/To4VFQ0OD7rvvPvXs2VMdO3ZU79699eCDD363781pAa3io+7Doa6uToMGDdLUqVN13XXXOT1O2JWVlcnr9WrYsGE6ffq07rnnHo0ePVoffPCBOnfu7PR4La5bt25auHChLrroIgUCAb344ou69tpr9d5776lfv35OjxdWlZWVeu655zRw4ECnRwm7fv36ad26dcH16Oj286/If//73xo+fLiuueYavfnmm+ratat2796tH/zgB06PFjaVlZVqaGgIru/cuVM/+clPdP311zs4Vfg8+uijWrx4sV588UX169dP27Zt05QpUxQfH6877rgj7PO0n3/6vkV2drays7OdHsMxJSUlIevLly9XUlKSqqqqNGLECIemCp/x48eHrD/88MNavHixNm/e3K4C5fjx48rNzdXzzz+vhx56yOlxwi46OvqsX7fRHjz66KNKS0vTsmXLgtt69uzp4ETh17Vr15D1hQsXqnfv3rr66qsdmii8Nm3apGuvvVY5OTmS/nNF6eWXX9bWrVsdmYeXeHBWtbW1kqTExESHJwm/hoYGrVy5UnV1de3uqxe8Xq9ycnKUlZXl9CiO2L17t1JTU9WrVy/l5uZq//79To8UNn/+8581dOhQXX/99UpKStLgwYP1/PPPOz2WY06ePKk//OEPmjp1arN/Ka1VV1xxhUpLS/XRRx9Jkv7+979r48aNjv3PO1dQcIbGxkbNnDlTw4cPV//+/Z0eJ2x27Nghj8ejEydO6Pzzz9eqVavUt29fp8cKm5UrV+rdd99VZWWl06M4IjMzU8uXL9cll1yiQ4cO6YEHHtBVV12lnTt3Ki4uzunxWtw///lPLV68WPn5+brnnntUWVmpO+64QzExMZo8ebLT44Xd6tWrdfToUd16661OjxI2c+fOld/vV0ZGhqKiotTQ0KCHH35Yubm5jsxDoOAMXq9XO3fu1MaNG50eJawuueQSbd++XbW1tfrjH/+oyZMnq6ysrF1EyoEDB3TnnXdq7dq16tChg9PjOOL//1/iwIEDlZmZqfT0dL366quaNm2ag5OFR2Njo4YOHapHHnlEkjR48GDt3LlTS5YsaZeB8sILLyg7O1upqalOjxI2r776qlasWKGioiL169dP27dv18yZM5WamurI3wECBSHy8vJUXFys8vJydevWzelxwiomJkYXXnihJGnIkCGqrKzUb3/7Wz333HMOT9byqqqqdPjwYf3oRz8KbmtoaFB5ebl+97vfqb6+XlFRUQ5OGH4JCQm6+OKLtWfPHqdHCYuUlJQzYrxPnz7605/+5NBEzvn444+1bt06vf76606PElazZ8/W3LlzNWnSJEnSgAED9PHHH6uwsJBAgXMCgYBuv/12rVq1Shs2bGh3N8edTWNjo+rr650eIyxGjRqlHTt2hGybMmWKMjIyNGfOnHYXJ9J/bhjeu3evbr75ZqdHCYvhw4ef8dECH330kdLT0x2ayDnLli1TUlJS8GbR9uKLL75QZGToralRUVFqbGx0ZB4C5b+OHz8e8n9K+/bt0/bt25WYmKju3bs7OFl4eL1eFRUV6Y033lBcXJx8Pp8kKT4+Xh07dnR4upZXUFCg7Oxsde/eXceOHVNRUZE2bNigt956y+nRwiIuLu6M+406d+6sLl26tJv7kO666y6NHz9e6enpOnjwoObPn6+oqCjdeOONTo8WFrNmzdIVV1yhRx55RL/4xS+0detWLV26VEuXLnV6tLBqbGzUsmXLNHny5Hb1NnPpP+9mfPjhh9W9e3f169dP7733np544glNnTrVmYECCAQCgcA777wTkHTGMnnyZKdHC4uznbukwLJly5weLSymTp0aSE9PD8TExAS6du0aGDVqVODtt992eixHXX311YE777zT6THC5oYbbgikpKQEYmJiAj/84Q8DN9xwQ2DPnj1OjxVWa9asCfTv3z8QGxsbyMjICCxdutTpkcLurbfeCkgKVFdXOz1K2Pn9/sCdd94Z6N69e6BDhw6BXr16BX7zm98E6uvrHZknIhBw6CPiAAAAvgafgwIAAMwhUAAAgDkECgAAMIdAAQAA5hAoAADAHAIFAACYQ6AAAABzCBQAAGAOgQIAAMwhUAAAgDkECgAAMIdAAQAA5vwfXsZMW8EHDcIAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 640x480 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#count the number of times each gene appears in the panels\n",
    "gene_count = {}\n",
    "for gene in gene_list:\n",
    "    gene_count[gene] = 0\n",
    "for panel in panel_list:\n",
    "    for gene in gene_list:\n",
    "        if gene in list(panel_dict[panel].columns):\n",
    "            gene_count[gene] += 1\n",
    "#plot a histogram of the number of times each gene appears in the panels\n",
    "import matplotlib.pyplot as plt\n",
    "plt.hist(list(gene_count.values()), bins=100)\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "225"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#get a list of the genes which occur in >=6 panels\n",
    "gene_list = []\n",
    "for gene in gene_count:\n",
    "    if gene_count[gene] >= 6:\n",
    "        gene_list.append(gene)\n",
    "len(gene_list)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#write gene_list to txt file\n",
    "with open('../data/crc_raw/cBioPortal_files/gene_list.txt', 'w') as f:\n",
    "    for gene in gene_list:\n",
    "        f.write(gene + '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(23225, 64)\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2684598/1285953628.py:1: DtypeWarning: Columns (17) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  muts = pd.read_csv('../data/crc_raw/cBioPortal_files/data_mutations_extended.txt', sep='\\t')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>...</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>Polyphen_Prediction</th>\n",
       "      <th>Polyphen_Score</th>\n",
       "      <th>SIFT_Prediction</th>\n",
       "      <th>SIFT_Score</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>Annotation_Status</th>\n",
       "      <th>mutationInCis_Flag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>PALB2</td>\n",
       "      <td>79728</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>16</td>\n",
       "      <td>23651599</td>\n",
       "      <td>23651599</td>\n",
       "      <td>+</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>Intron</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>113.0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>FBXW7</td>\n",
       "      <td>55294</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>4</td>\n",
       "      <td>153258956</td>\n",
       "      <td>153258956</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>0.993</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>230.0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>1956</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>55221710</td>\n",
       "      <td>55221710</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>possibly_damaging</td>\n",
       "      <td>0.882</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.03</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>246.0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PSMD13</td>\n",
       "      <td>5719</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>11</td>\n",
       "      <td>247403</td>\n",
       "      <td>247403</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>possibly_damaging</td>\n",
       "      <td>0.811</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>0.08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>301.0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>3845</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>25398284</td>\n",
       "      <td>25398284</td>\n",
       "      <td>+</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>...</td>\n",
       "      <td>PASS</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.303</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>0.00</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>277.0</td>\n",
       "      <td>SUCCESS</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 64 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position   \n",
       "0       PALB2           79728   DFCI     GRCh37         16        23651599  \\\n",
       "1       FBXW7           55294   DFCI     GRCh37          4       153258956   \n",
       "2        EGFR            1956   DFCI     GRCh37          7        55221710   \n",
       "3      PSMD13            5719   DFCI     GRCh37         11          247403   \n",
       "4        KRAS            3845   DFCI     GRCh37         12        25398284   \n",
       "\n",
       "   End_Position Strand                             Consequence   \n",
       "0      23651599      +                          intron_variant  \\\n",
       "1     153258956      +  missense_variant,splice_region_variant   \n",
       "2      55221710      +                        missense_variant   \n",
       "3        247403      +                        missense_variant   \n",
       "4      25398284      +                        missense_variant   \n",
       "\n",
       "  Variant_Classification  ... FILTER Polyphen_Prediction Polyphen_Score   \n",
       "0                 Intron  ...   PASS                 NaN            NaN  \\\n",
       "1      Missense_Mutation  ...   PASS   probably_damaging          0.993   \n",
       "2      Missense_Mutation  ...   PASS   possibly_damaging          0.882   \n",
       "3      Missense_Mutation  ...   PASS   possibly_damaging          0.811   \n",
       "4      Missense_Mutation  ...   PASS              benign          0.303   \n",
       "\n",
       "  SIFT_Prediction SIFT_Score  SWISSPROT n_depth t_depth  Annotation_Status   \n",
       "0             NaN        NaN        NaN     NaN   113.0            SUCCESS  \\\n",
       "1     deleterious       0.00        NaN     NaN   230.0            SUCCESS   \n",
       "2     deleterious       0.03        NaN     NaN   246.0            SUCCESS   \n",
       "3       tolerated       0.08        NaN     NaN   301.0            SUCCESS   \n",
       "4     deleterious       0.00        NaN     NaN   277.0            SUCCESS   \n",
       "\n",
       "   mutationInCis_Flag  \n",
       "0               False  \n",
       "1               False  \n",
       "2               False  \n",
       "3               False  \n",
       "4               False  \n",
       "\n",
       "[5 rows x 64 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "muts = pd.read_csv('../data/crc_raw/cBioPortal_files/data_mutations_extended.txt', sep='\\t')\n",
    "print(muts.shape)\n",
    "muts.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['Hugo_Symbol', 'Entrez_Gene_Id', 'Center', 'NCBI_Build', 'Chromosome',\n",
       "       'Start_Position', 'End_Position', 'Strand', 'Consequence',\n",
       "       'Variant_Classification', 'Variant_Type', 'Reference_Allele',\n",
       "       'Tumor_Seq_Allele1', 'Tumor_Seq_Allele2', 'dbSNP_RS',\n",
       "       'dbSNP_Val_Status', 'Tumor_Sample_Barcode',\n",
       "       'Matched_Norm_Sample_Barcode', 'Match_Norm_Seq_Allele1',\n",
       "       'Match_Norm_Seq_Allele2', 'Tumor_Validation_Allele1',\n",
       "       'Tumor_Validation_Allele2', 'Match_Norm_Validation_Allele1',\n",
       "       'Match_Norm_Validation_Allele2', 'Verification_Status',\n",
       "       'Validation_Status', 'Mutation_Status', 'Sequencing_Phase',\n",
       "       'Sequence_Source', 'Validation_Method', 'Score', 'BAM_File',\n",
       "       'Sequencer', 't_ref_count', 't_alt_count', 'n_ref_count', 'n_alt_count',\n",
       "       'HGVSc', 'HGVSp', 'HGVSp_Short', 'Transcript_ID', 'RefSeq',\n",
       "       'Protein_position', 'Codons', 'Exon_Number', 'gnomAD_AF',\n",
       "       'gnomAD_AFR_AF', 'gnomAD_AMR_AF', 'gnomAD_ASJ_AF', 'gnomAD_EAS_AF',\n",
       "       'gnomAD_FIN_AF', 'gnomAD_NFE_AF', 'gnomAD_OTH_AF', 'gnomAD_SAS_AF',\n",
       "       'FILTER', 'Polyphen_Prediction', 'Polyphen_Score', 'SIFT_Prediction',\n",
       "       'SIFT_Score', 'SWISSPROT', 'n_depth', 't_depth', 'Annotation_Status',\n",
       "       'mutationInCis_Flag'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "muts.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Oncotree Code</th>\n",
       "      <th>Sample Type</th>\n",
       "      <th>Sequence Assay ID</th>\n",
       "      <th>Age at sequencing</th>\n",
       "      <th>Sample Identifier</th>\n",
       "      <th>Patient Identifier</th>\n",
       "      <th>Mismatch Repair (MMR) Testing at Time of Sample Acquisition</th>\n",
       "      <th>MSI-H test result at Time of Sample Acquisition</th>\n",
       "      <th>Microsatellite Instability (MSI) Testing at Time of Sample Acquisition</th>\n",
       "      <th>Positive PD-L1 Result at the Time of Sample Acquisition</th>\n",
       "      <th>PD-L1 Testing at Time of Sample Acquisition</th>\n",
       "      <th>Year of Next Generation Sequencing</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>#The OncoTree diagnosis code using the abbrevi...</td>\n",
       "      <td>The type of sample (i.e., normal, primary, met...</td>\n",
       "      <td>Sequence Assay ID</td>\n",
       "      <td>Age at sequencing</td>\n",
       "      <td>A unique sample identifier.</td>\n",
       "      <td>Identifier to uniquely specify a patient.</td>\n",
       "      <td>Indicates whether any mismatch repair (MMR) te...</td>\n",
       "      <td>Indicates whether any Microsatellite Instabili...</td>\n",
       "      <td>Indicates whether any microsatellite instabili...</td>\n",
       "      <td>Indicates whether any PD-L1 testing on the pat...</td>\n",
       "      <td>Indicates whether any PD-L1 testing was report...</td>\n",
       "      <td>Year of Next Generation Sequencing</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>#STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>NUMBER</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "      <td>STRING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>#1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ONCOTREE_CODE</td>\n",
       "      <td>SAMPLE_TYPE_DETAILED</td>\n",
       "      <td>SEQ_ASSAY_ID</td>\n",
       "      <td>AGE_AT_SEQUENCING</td>\n",
       "      <td>SAMPLE_ID</td>\n",
       "      <td>PATIENT_ID</td>\n",
       "      <td>MMR_TESTING</td>\n",
       "      <td>MSI_HIGH_ANY</td>\n",
       "      <td>MSI_TESTING</td>\n",
       "      <td>PDL1_POSITIVE_ANY</td>\n",
       "      <td>PDL1_TESTING</td>\n",
       "      <td>CPT_SEQ_DATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>COADREAD</td>\n",
       "      <td>Metastasis site unspecified</td>\n",
       "      <td>DFCI-ONCOPANEL-3</td>\n",
       "      <td>53</td>\n",
       "      <td>GENIE-DFCI-000147-350443</td>\n",
       "      <td>GENIE-DFCI-000147</td>\n",
       "      <td>Yes</td>\n",
       "      <td>NaN</td>\n",
       "      <td>No</td>\n",
       "      <td>Yes</td>\n",
       "      <td>Yes</td>\n",
       "      <td>2017</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      #Oncotree Code   \n",
       "0  #The OncoTree diagnosis code using the abbrevi...  \\\n",
       "1                                            #STRING   \n",
       "2                                                 #1   \n",
       "3                                      ONCOTREE_CODE   \n",
       "4                                           COADREAD   \n",
       "\n",
       "                                         Sample Type  Sequence Assay ID   \n",
       "0  The type of sample (i.e., normal, primary, met...  Sequence Assay ID  \\\n",
       "1                                             STRING             STRING   \n",
       "2                                                  1                  1   \n",
       "3                               SAMPLE_TYPE_DETAILED       SEQ_ASSAY_ID   \n",
       "4                        Metastasis site unspecified   DFCI-ONCOPANEL-3   \n",
       "\n",
       "   Age at sequencing            Sample Identifier   \n",
       "0  Age at sequencing  A unique sample identifier.  \\\n",
       "1             NUMBER                       STRING   \n",
       "2                  1                            1   \n",
       "3  AGE_AT_SEQUENCING                    SAMPLE_ID   \n",
       "4                 53     GENIE-DFCI-000147-350443   \n",
       "\n",
       "                          Patient Identifier   \n",
       "0  Identifier to uniquely specify a patient.  \\\n",
       "1                                     STRING   \n",
       "2                                          1   \n",
       "3                                 PATIENT_ID   \n",
       "4                          GENIE-DFCI-000147   \n",
       "\n",
       "  Mismatch Repair (MMR) Testing at Time of Sample Acquisition   \n",
       "0  Indicates whether any mismatch repair (MMR) te...           \\\n",
       "1                                             STRING            \n",
       "2                                                  1            \n",
       "3                                        MMR_TESTING            \n",
       "4                                                Yes            \n",
       "\n",
       "     MSI-H test result at Time of Sample Acquisition   \n",
       "0  Indicates whether any Microsatellite Instabili...  \\\n",
       "1                                             STRING   \n",
       "2                                                  1   \n",
       "3                                       MSI_HIGH_ANY   \n",
       "4                                                NaN   \n",
       "\n",
       "  Microsatellite Instability (MSI) Testing at Time of Sample Acquisition   \n",
       "0  Indicates whether any microsatellite instabili...                      \\\n",
       "1                                             STRING                       \n",
       "2                                                  1                       \n",
       "3                                        MSI_TESTING                       \n",
       "4                                                 No                       \n",
       "\n",
       "  Positive PD-L1 Result at the Time of Sample Acquisition   \n",
       "0  Indicates whether any PD-L1 testing on the pat...       \\\n",
       "1                                             STRING        \n",
       "2                                                  1        \n",
       "3                                  PDL1_POSITIVE_ANY        \n",
       "4                                                Yes        \n",
       "\n",
       "         PD-L1 Testing at Time of Sample Acquisition   \n",
       "0  Indicates whether any PD-L1 testing was report...  \\\n",
       "1                                             STRING   \n",
       "2                                                  1   \n",
       "3                                       PDL1_TESTING   \n",
       "4                                                Yes   \n",
       "\n",
       "   Year of Next Generation Sequencing  \n",
       "0  Year of Next Generation Sequencing  \n",
       "1                              STRING  \n",
       "2                                   1  \n",
       "3                        CPT_SEQ_DATE  \n",
       "4                                2017  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "samps = pd.read_csv('../data/crc_raw/cBioPortal_files/data_clinical_sample.txt', sep='\\t')\n",
    "samps.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cohort</th>\n",
       "      <th>record_id</th>\n",
       "      <th>institution</th>\n",
       "      <th>cpt_number</th>\n",
       "      <th>ca_seq</th>\n",
       "      <th>cpt_n_ca_seq</th>\n",
       "      <th>cpt_order_int</th>\n",
       "      <th>cpt_seq_date</th>\n",
       "      <th>dob_cpt_report_days</th>\n",
       "      <th>dob_cpt_report_mos</th>\n",
       "      <th>...</th>\n",
       "      <th>path_proc_cpt_rep_days</th>\n",
       "      <th>path_proc_cpt_rep_mos</th>\n",
       "      <th>path_proc_cpt_rep_yrs</th>\n",
       "      <th>path_proc_number</th>\n",
       "      <th>path_rep_number</th>\n",
       "      <th>cpt_genie_sample_id</th>\n",
       "      <th>cpt_oncotree_code</th>\n",
       "      <th>sample_type</th>\n",
       "      <th>cpt_seq_assay_id</th>\n",
       "      <th>release_version</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>CRC</td>\n",
       "      <td>GENIE-DFCI-000048</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>22066</td>\n",
       "      <td>2015</td>\n",
       "      <td>22108</td>\n",
       "      <td>727.236842</td>\n",
       "      <td>...</td>\n",
       "      <td>394</td>\n",
       "      <td>12.960526</td>\n",
       "      <td>1.078713</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>GENIE-DFCI-000048-7508</td>\n",
       "      <td>COAD</td>\n",
       "      <td>Primary tumor</td>\n",
       "      <td>DFCI-ONCOPANEL-2</td>\n",
       "      <td>2.0-public</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CRC</td>\n",
       "      <td>GENIE-DFCI-000147</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>18731</td>\n",
       "      <td>2015</td>\n",
       "      <td>18775</td>\n",
       "      <td>617.598684</td>\n",
       "      <td>...</td>\n",
       "      <td>121</td>\n",
       "      <td>3.980263</td>\n",
       "      <td>0.331280</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>GENIE-DFCI-000147-6898</td>\n",
       "      <td>READ</td>\n",
       "      <td>Primary tumor</td>\n",
       "      <td>DFCI-ONCOPANEL-2</td>\n",
       "      <td>2.0-public</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CRC</td>\n",
       "      <td>GENIE-DFCI-000147</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>19476</td>\n",
       "      <td>2017</td>\n",
       "      <td>19488</td>\n",
       "      <td>641.052632</td>\n",
       "      <td>...</td>\n",
       "      <td>33</td>\n",
       "      <td>1.085526</td>\n",
       "      <td>0.090349</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>GENIE-DFCI-000147-350443</td>\n",
       "      <td>COADREAD</td>\n",
       "      <td>Metastasis site unspecified</td>\n",
       "      <td>DFCI-ONCOPANEL-3</td>\n",
       "      <td>2.0-public</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CRC</td>\n",
       "      <td>GENIE-DFCI-000233</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>20356</td>\n",
       "      <td>2015</td>\n",
       "      <td>20399</td>\n",
       "      <td>671.019737</td>\n",
       "      <td>...</td>\n",
       "      <td>2038</td>\n",
       "      <td>67.039474</td>\n",
       "      <td>5.579740</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>GENIE-DFCI-000233-7633</td>\n",
       "      <td>COAD</td>\n",
       "      <td>Metastasis site unspecified</td>\n",
       "      <td>DFCI-ONCOPANEL-2</td>\n",
       "      <td>2.0-public</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CRC</td>\n",
       "      <td>GENIE-DFCI-000247</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>23940</td>\n",
       "      <td>2016</td>\n",
       "      <td>23956</td>\n",
       "      <td>788.026316</td>\n",
       "      <td>...</td>\n",
       "      <td>597</td>\n",
       "      <td>19.638158</td>\n",
       "      <td>1.634497</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>GENIE-DFCI-000247-11259</td>\n",
       "      <td>READ</td>\n",
       "      <td>Primary tumor</td>\n",
       "      <td>DFCI-ONCOPANEL-2</td>\n",
       "      <td>2.0-public</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  cohort          record_id institution  cpt_number  ca_seq  cpt_n_ca_seq   \n",
       "0    CRC  GENIE-DFCI-000048        DFCI           1       0             1  \\\n",
       "1    CRC  GENIE-DFCI-000147        DFCI           1       0             1   \n",
       "2    CRC  GENIE-DFCI-000147        DFCI           2       0             1   \n",
       "3    CRC  GENIE-DFCI-000233        DFCI           1       2             1   \n",
       "4    CRC  GENIE-DFCI-000247        DFCI           1       1             1   \n",
       "\n",
       "   cpt_order_int  cpt_seq_date  dob_cpt_report_days  dob_cpt_report_mos  ...   \n",
       "0          22066          2015                22108          727.236842  ...  \\\n",
       "1          18731          2015                18775          617.598684  ...   \n",
       "2          19476          2017                19488          641.052632  ...   \n",
       "3          20356          2015                20399          671.019737  ...   \n",
       "4          23940          2016                23956          788.026316  ...   \n",
       "\n",
       "   path_proc_cpt_rep_days  path_proc_cpt_rep_mos  path_proc_cpt_rep_yrs   \n",
       "0                     394              12.960526               1.078713  \\\n",
       "1                     121               3.980263               0.331280   \n",
       "2                      33               1.085526               0.090349   \n",
       "3                    2038              67.039474               5.579740   \n",
       "4                     597              19.638158               1.634497   \n",
       "\n",
       "   path_proc_number  path_rep_number       cpt_genie_sample_id   \n",
       "0                 1                2    GENIE-DFCI-000048-7508  \\\n",
       "1                 2                1    GENIE-DFCI-000147-6898   \n",
       "2                 5                1  GENIE-DFCI-000147-350443   \n",
       "3                 4                1    GENIE-DFCI-000233-7633   \n",
       "4                 1                1   GENIE-DFCI-000247-11259   \n",
       "\n",
       "   cpt_oncotree_code                  sample_type  cpt_seq_assay_id   \n",
       "0               COAD                Primary tumor  DFCI-ONCOPANEL-2  \\\n",
       "1               READ                Primary tumor  DFCI-ONCOPANEL-2   \n",
       "2           COADREAD  Metastasis site unspecified  DFCI-ONCOPANEL-3   \n",
       "3               COAD  Metastasis site unspecified  DFCI-ONCOPANEL-2   \n",
       "4               READ                Primary tumor  DFCI-ONCOPANEL-2   \n",
       "\n",
       "   release_version  \n",
       "0       2.0-public  \n",
       "1       2.0-public  \n",
       "2       2.0-public  \n",
       "3       2.0-public  \n",
       "4       2.0-public  \n",
       "\n",
       "[5 rows x 29 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "panels = pd.read_csv('../data/crc_raw/CRC_2.0-public_clinical_data/cancer_panel_test_level_dataset.csv')\n",
    "panels.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['cohort', 'record_id', 'institution', 'cpt_number', 'ca_seq',\n",
       "       'cpt_n_ca_seq', 'cpt_order_int', 'cpt_seq_date', 'dob_cpt_report_days',\n",
       "       'dob_cpt_report_mos', 'dob_cpt_report_yrs', 'cpt_report_post_death',\n",
       "       'cpt_report_post_last_alive', 'dx_cpt_rep_days', 'dx_cpt_rep_mos',\n",
       "       'dx_cpt_rep_yrs', 'dx_path_proc_cpt_days', 'dx_path_proc_cpt_mos',\n",
       "       'dx_path_proc_cpt_yrs', 'path_proc_cpt_rep_days',\n",
       "       'path_proc_cpt_rep_mos', 'path_proc_cpt_rep_yrs', 'path_proc_number',\n",
       "       'path_rep_number', 'cpt_genie_sample_id', 'cpt_oncotree_code',\n",
       "       'sample_type', 'cpt_seq_assay_id', 'release_version'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "panels.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15612, 64)\n"
     ]
    }
   ],
   "source": [
    "#keep only the rows where 'Hugo_Symbol' is in gene_list\n",
    "muts = muts[muts['Hugo_Symbol'].isin(gene_list)]\n",
    "print(muts.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1531, 225)\n"
     ]
    }
   ],
   "source": [
    "mut_df = pd.DataFrame(columns=gene_list, index=muts['Tumor_Sample_Barcode'].unique())\n",
    "print(mut_df.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CDK4</th>\n",
       "      <th>CCND3</th>\n",
       "      <th>CDH1</th>\n",
       "      <th>CDK8</th>\n",
       "      <th>GNAS</th>\n",
       "      <th>PRKAR1A</th>\n",
       "      <th>MLH1</th>\n",
       "      <th>PNRC1</th>\n",
       "      <th>ETV1</th>\n",
       "      <th>TSC1</th>\n",
       "      <th>...</th>\n",
       "      <th>ROS1</th>\n",
       "      <th>DAXX</th>\n",
       "      <th>TSC2</th>\n",
       "      <th>RAD51C</th>\n",
       "      <th>KMT2D</th>\n",
       "      <th>ARID1A</th>\n",
       "      <th>AURKB</th>\n",
       "      <th>U2AF1</th>\n",
       "      <th>REL</th>\n",
       "      <th>BRCA1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GENIE-DFCI-002643-6598</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENIE-DFCI-008624-7126</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENIE-DFCI-003423-6929</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENIE-DFCI-007675-6942</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GENIE-DFCI-008889-6970</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 225 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                        CDK4  CCND3  CDH1  CDK8  GNAS  PRKAR1A  MLH1  PNRC1   \n",
       "GENIE-DFCI-002643-6598     0      0     0     0     0        0     0      0  \\\n",
       "GENIE-DFCI-008624-7126     0      0     0     0     0        0     0      0   \n",
       "GENIE-DFCI-003423-6929     0      0     1     0     0        0     0      0   \n",
       "GENIE-DFCI-007675-6942     0      0     0     0     0        0     0      0   \n",
       "GENIE-DFCI-008889-6970     0      0     0     0     0        0     0      0   \n",
       "\n",
       "                        ETV1  TSC1  ...  ROS1  DAXX  TSC2  RAD51C  KMT2D   \n",
       "GENIE-DFCI-002643-6598     0     0  ...     1     0     0       0      0  \\\n",
       "GENIE-DFCI-008624-7126     0     0  ...     0     0     0       0      0   \n",
       "GENIE-DFCI-003423-6929     0     0  ...     0     0     0       0      0   \n",
       "GENIE-DFCI-007675-6942     0     0  ...     0     0     0       0      1   \n",
       "GENIE-DFCI-008889-6970     0     0  ...     0     0     0       0      0   \n",
       "\n",
       "                        ARID1A  AURKB  U2AF1  REL  BRCA1  \n",
       "GENIE-DFCI-002643-6598       0      0      0    0      0  \n",
       "GENIE-DFCI-008624-7126       0      0      0    0      0  \n",
       "GENIE-DFCI-003423-6929       0      0      0    0      0  \n",
       "GENIE-DFCI-007675-6942       0      0      0    0      0  \n",
       "GENIE-DFCI-008889-6970       0      0      0    0      0  \n",
       "\n",
       "[5 rows x 225 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for i in range(muts.shape[0]):\n",
    "    mut_df.loc[muts.iloc[i]['Tumor_Sample_Barcode'], muts.iloc[i]['Hugo_Symbol']] = 1\n",
    "mut_df = mut_df.fillna(0)\n",
    "mut_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mut_df.to_csv('../data/crc_mutpertreat.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(321, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>record_id</th>\n",
       "      <th>institution</th>\n",
       "      <th>dx_drug_start_int_1</th>\n",
       "      <th>os_g_status</th>\n",
       "      <th>pfs_i_and_m_g_status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>GENIE-DFCI-000971</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>795</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>GENIE-DFCI-001038</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>759</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>GENIE-DFCI-002507</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>2142</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>129</th>\n",
       "      <td>GENIE-DFCI-002561</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>1812</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>250</th>\n",
       "      <td>GENIE-DFCI-005003</td>\n",
       "      <td>DFCI</td>\n",
       "      <td>946</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             record_id institution  dx_drug_start_int_1  os_g_status   \n",
       "37   GENIE-DFCI-000971        DFCI                  795            1  \\\n",
       "41   GENIE-DFCI-001038        DFCI                  759            1   \n",
       "108  GENIE-DFCI-002507        DFCI                 2142            1   \n",
       "129  GENIE-DFCI-002561        DFCI                 1812            1   \n",
       "250  GENIE-DFCI-005003        DFCI                  946            1   \n",
       "\n",
       "     pfs_i_and_m_g_status  \n",
       "37                    1.0  \n",
       "41                    1.0  \n",
       "108                   1.0  \n",
       "129                   1.0  \n",
       "250                   1.0  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ib_out = pd.read_csv('../data/crc_ib_out.csv', index_col=0)\n",
    "print(ib_out.shape)\n",
    "ib_out.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "muts = pd.read_csv('../data/')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
